WordPress/src/main/java/org/wordpress/android/ui/posts/adapters/AuthorSelectionAdapter.kt

```diff
@@ -0,0 +1,209 @@
+package org.wordpress.android.ui.posts.adapters
+
+import android.content.Context
+import android.database.DataSetObserver
+import android.support.annotation.ColorRes
+import android.support.annotation.DrawableRes
+import android.support.v4.content.ContextCompat
+import android.support.v7.widget.AppCompatImageView
+import android.support.v7.widget.AppCompatTextView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.SpinnerAdapter
+import androidx.annotation.StringRes
+import org.wordpress.android.R
+import org.wordpress.android.WordPress
+import org.wordpress.android.util.DisplayUtils
+import org.wordpress.android.util.GravatarUtils
+import org.wordpress.android.util.image.ImageManager
+import org.wordpress.android.util.image.ImageType.AVATAR_WITH_BACKGROUND
+import java.lang.ref.WeakReference
+import javax.inject.Inject
+
+class AuthorSelectionAdapter(context: Context) : SpinnerAdapter {
```

Comment:
We recently started using a pattern which tries to move all the logic out of the adapters/ViewHolders to ViewModels. 

Look for example at these
https://github.com/wordpress-mobile/WordPress-Android/blob/d724c78acce2da2bd7f561c281205d6f749c92e4/WordPress/src/main/java/org/wordpress/android/ui/sitecreation/domains/NewSiteCreationDomainsAdapter.kt

https://github.com/wordpress-mobile/WordPress-Android/blob/d724c78acce2da2bd7f561c281205d6f749c92e4/WordPress/src/main/java/org/wordpress/android/ui/sitecreation/domains/NewSiteCreationDomainViewHolder.kt

The idea is to create a ViewState in the ViewModel. ViewState contains all the data for the View - data, string resources, selected state etc. The main benefit of this approach is we can write unit tests for the ViewModel and there is almost no need to write tests for the view as it doesn't contain any real logic.

Since this is basically a new code I think we should consider using the same pattern. Wdyt? I'd love to hear your thoughts on the pattern:)!